lib/account.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+require 'time'
+
+module Bank
+  class Account
+    # attr_accessor :withdraw, :deposit, :all_accounts
+    attr_reader :id, :balance, :opendate, :show_id, :account_count, :find_account
+    @@count = 0
+
+    def initialize(id, balance, opendate)
+      @@count += 1
+      @id = id.to_i
+      @balance = monify(balance)
+      unless @balance >= 0
+        raise ArgumentError.new "Starting balance is not valid."
+      end
+      @opendate = Time.parse(opendate)
+      @withdraw_fee = 0
+      @minimum_balance = 0
+      # @opendate = DateTime.strptime(opendate, '%Y-%m-%d %H:%M:%S %z')
+    end
+
+    # check if balance is in cents or in dollars with '.'
+    def monify(amount)
+      float_already = false
+      amount.to_s.each_char do |letter|
+        if letter == "."
+          float_already = true
+        else
+        end
+      end
+      if float_already == false
+        return amount.to_i/100.0
+      else
+        return amount.to_f
+      end
+    end
+
+    def withdraw(amount)
```

Comment:
I really like the way you've used instance variables here to unify the logic between the different account types. Good work!